seatunnel-connectors-v2/connector-http/connector-http-base/src/test/java/org/apache/seatunnel/connectors/seatunnel/http/HttpSourceReaderUpdateRequestParamTest.java

```diff
@@ -0,0 +1,366 @@
+import java.util.Map;
+
+import static org.mockito.ArgumentMatchers.any;
+import static org.mockito.ArgumentMatchers.anyBoolean;
+import static org.mockito.ArgumentMatchers.anyString;
+import static org.mockito.Mockito.when;
+
+public class HttpSourceReaderUpdateRequestParamTest {
+
+    private HttpParameter httpParameter;
+    private JsonField jsonField;
+    private SimpleTextDeserializationSchema deserializationSchema;
+    private HttpSourceReader httpSourceReader;
+
+    @Mock private SingleSplitReaderContext context;
+
+    @Mock private Collector<SeaTunnelRow> collector;
+
+    @Mock private HttpClientProvider httpClientProvider;
+
+    @Mock private HttpResponse httpResponse;
+
+    @BeforeEach
+    public void setUp() throws Exception {
+        MockitoAnnotations.openMocks(this);
+
+        httpParameter = new HttpParameter();
+        httpParameter.setUrl("http://test-url.com");
+        httpParameter.setMethod(HttpRequestMethod.GET);
+
+        Map<String, String> fields = new HashMap<>();
+        fields.put("key1", "$[*].key1");
+        fields.put("key2", "$[*].key2");
+        jsonField = JsonField.builder().fields(fields).build();
+
+        // Create the schema with two string fields
+        SeaTunnelRowType rowType =
+                new SeaTunnelRowType(
+                        new String[] {"key1", "key2"},
+                        new SeaTunnelDataType[] {BasicType.STRING_TYPE, BasicType.STRING_TYPE});
+        deserializationSchema = new SimpleTextDeserializationSchema(rowType);
+
+        // Setup mocks
+        when(httpResponse.getCode()).thenReturn(200);
+        when(collector.getCheckpointLock()).thenReturn(new Object());
+        when(httpClientProvider.execute(
+                        anyString(), anyString(), any(), any(), any(), anyBoolean()))
+                .thenReturn(httpResponse);
+
+        // Create HttpSourceReader
+        httpSourceReader =
+                new HttpSourceReader(
+                        httpParameter, context, deserializationSchema, jsonField, null);
+
+        httpSourceReader.open();
+    }
+
+    @Test
+    public void testUpdateRequestParamWithHeaderPlaceholder() throws Exception {
+        // Setup test data
+        Map<String, String> headers = new HashMap<>();
+        headers.put("Page-Number", "${page}");
+        headers.put("Authorization", "Bearer token-123");
+        httpParameter.setHeaders(headers);
+
+        PageInfo pageInfo = new PageInfo();
+        pageInfo.setPageField("page");
+        pageInfo.setPageIndex(5L);
+        pageInfo.setUsePlaceholderReplacement(true);
+
+        // Call updateRequestParam method using reflection
+        Method updateRequestParamMethod =
+                HttpSourceReader.class.getDeclaredMethod(
+                        "updateRequestParam", PageInfo.class, boolean.class);
+        updateRequestParamMethod.setAccessible(true);
+        updateRequestParamMethod.invoke(httpSourceReader, pageInfo, true);
```

Comment:
Update `updateRequestParam` scope to default and add `@VisibleForTesting` annotation to avoid reflection.